gpkit/constraints/set.py

```diff
@@ -177,9 +180,9 @@ def as_gpconstr(self, x0):
         """Returns GPConstraint approximating this constraint at x0
 
         When x0 is none, may return a default guess."""
-        cs = ConstraintSet([constr.as_gpconstr(x0) for constr in self])
-        cs.substitutions.update(self.substitutions)
-        return cs
+        gpconstrs = [constr.as_gpconstr(x0) for constr in self]
+        return ConstraintSet(gpconstrs,
+                             self.substitutions, recursesubs=False)
```

Comment:
They've been set to the model's substitutions dict... by `ConstraintSet.__init__`? The user can change the top-level model's substitutions, but not substitutions of submodels, correct?